DllImportGenerator/DllImportGenerator.UnitTests/DllImportGeneratorOptionsProvider.cs

```diff
@@ -0,0 +1,66 @@
+using System.Diagnostics.CodeAnalysis;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.Diagnostics;
+
+namespace DllImportGenerator.UnitTests
+{
+    internal class DllImportGeneratorOptionsProvider : AnalyzerConfigOptionsProvider
+    {
+        public DllImportGeneratorOptionsProvider(bool useMarshalType, bool generateForwarders)
+        {
+            GlobalOptions = new GlobalGeneratorOptions(useMarshalType, generateForwarders);
+        }
+
+        public override AnalyzerConfigOptions GlobalOptions  { get; }
+
+        public override AnalyzerConfigOptions GetOptions(SyntaxTree tree)
+        {
+            return EmptyOptions.Instance;
+        }
+
+        public override AnalyzerConfigOptions GetOptions(AdditionalText textFile)
+        {
+            return EmptyOptions.Instance;
+        }
+
+        class GlobalGeneratorOptions : AnalyzerConfigOptions
+        {
+            private readonly bool _useMarshalType = false;
```

Comment:
I think we are keeping with the `this.` syntax rather than the `_`.

Reply:
Our code base is very much a mix depending on who wrote each part. Parts that I wrote have the `_` prefix and parts that you and Elinor wrote have the `this.` prefix. Personally I prefer not using either prefix and I really don't like the `this.` prefix, it feels particularly clunky to me. I adjusted to the `_` prefix because people wanted a prefix and dotnet/runtime uses the `_` prefix.

After the holiday break we should set up an .editorconfig file so we can enforce the style rules we decide on as part of the build.

Reply:
yep that makes sense. @elinor-fung and I are used to the old StyleCop rules and follow them because it makes code reviews easier for us. I agree we should have the conversation.

Reply:
I'm used to the default C# style options in VS, so that's what I default to. Once we have the editorconfig, Roslyn will help out with enforcing our style, so I'm willing to be convinced to change styles. We can also install the Rosyln StyleCop package to enforce the rules if we go that route.